packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -6278,50 +6281,197 @@ abstract class RenderObjectElement extends Element {
     return ancestor as RenderObjectElement?;
   }
 
-  ParentDataElement<ParentData>? _findAncestorParentDataElement() {
-    Element? ancestor = _parent;
-    ParentDataElement<ParentData>? result;
-    while (ancestor != null && ancestor is! RenderObjectElement) {
-      if (ancestor is ParentDataElement<ParentData>) {
-        result = ancestor;
-        break;
-      }
-      ancestor = ancestor._parent;
-    }
+  void _debugCheckCompetingAncestors(
+    List<ParentDataElement<ParentData>> result,
+    Set<Type> debugAncestorTypes,
+    List<Type> debugAncestorCulprits,
+  ) {
     assert(() {
-      if (result == null || ancestor == null) {
-        return true;
+      List<ParentDataElement<ParentData>> getCompetingAncestors() {
+        return result.where((ParentDataElement<ParentData> ancestor) {
+          return debugAncestorCulprits.contains(ancestor.runtimeType);
+        }).toList();
```

Comment:
Why not just 

```suggestion
      Iterable<ParentDataElement<ParentData>> competingAncestors = result.where((ParentDataElement<ParentData> ancestor) {
          return debugAncestorCulprits.contains(ancestor.runtimeType);
        });
```

Reply:
And move it into the `if`.

Reply:
I wasn't sure if putting it in the `if` was unfriendly for readability, I gave it a try with latest commit.